src/mlpack/methods/gmm/em_fit_impl.hpp

```diff
@@ -361,7 +399,13 @@ ArmadilloGMMWrapper(const arma::mat& observations,
     for (size_t i = 0; i < dists.size(); ++i)
     {
       means.col(i) = dists[i].Mean();
-      covs.col(i) = dists[i].Covariance().diag();
+
+      // DiagonalGaussianDistribution has diagonal covariance as an arma::vec.
+      if (std::is_same<Distribution,
+          distribution::DiagonalGaussianDistribution>::value)
+        covs.col(i) = dists[i].Covariance();
+      else
+        covs.col(i) = dists[i].Covariance().diag();
```

Comment:
With the code change I suggested above, this would always be called with `DiagonalGaussianDistribution`, so we could simplify this code a little bit.

Reply:
I have a suggestion above, see what you think of that. :+1: